src/transaction/input.rs

```diff
@@ -73,19 +33,34 @@ impl BlindedInput {
         self.dbc_id
     }
 
-    /// Verify if a blinded amount you know of, is the same as the one in the input,
-    /// and that the bytes passed in, are what the signature of this input was made over,
-    /// and that the public key of this input was the signer.
-    pub fn verify(&self, msg: &[u8], blinded_amount: BlindedAmount) -> Result<()> {
-        if self.blinded_amount != blinded_amount {
-            return Err(Error::InvalidInputBlindedAmount);
-        }
-
-        // check the signature
+    /// Verify the Signature
+    pub fn verify(&self, msg: &[u8]) -> Result<()> {
         if !self.dbc_id.verify(&self.signature, msg) {
             return Err(Error::InvalidSignature);
         }
 
         Ok(())
     }
 }
+
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Debug, Clone)]
+pub struct InputIntermediate {
```

Comment:
Just brainstorming here, any chance we could get rid of this extra type and only have one type for `Input` in the API? 
I would simplify our exposed API as we'd only have one input type. 
Just wondering here, and also probably out of scope for this PR.

Reply:
Removing `Input::signature` might actually allow us to remove this extra struct. And I think it would be safe to remove the `signature` field? It has been briefed in the above comment.